Bottom: 410819dc2c406c3bffdc4976599da7178b0174e4
Top:    aa14305a639cc24bc37e83f72d0556816fd34ba2
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-24 20:05:13 +0000

polygon.c: Temporarily disable coalescing adjacent line segments

We need to extend the tests to properly account for arc segments.


---

diff --git a/src/polygon.c b/src/polygon.c
index e58f7db..0395ac0 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1383,6 +1383,7 @@ RemoveExcessPolygonPoints (LayerType *Layer, PolygonType *Polygon)
   if (Undoing ())
     return (false);
 
+if (0) {
   for (n = 0; n < Polygon->PointN; n++)
     {
       prev = prev_contour_point (Polygon, n);
@@ -1392,6 +1393,7 @@ RemoveExcessPolygonPoints (LayerType *Layer, PolygonType *Polygon)
       line.Point1 = Polygon->Points[prev];
       line.Point2 = Polygon->Points[next];
       line.Thickness = 0;
+#warning THIS TEST IS NOT SUFFICIENT.. DOESNT ACCOUNT FOR DIFFERENT ARC CENTERS
       if (Polygon->Points[prev].included_angle == Polygon->Points[n].included_angle &&
           IsPointOnLine (p->X, p->Y, 0.0, &line))
         {
@@ -1399,6 +1401,7 @@ RemoveExcessPolygonPoints (LayerType *Layer, PolygonType *Polygon)
           changed = true;
         }
     }
+}
   return (changed);
 }
